lib/internal/Magento/Framework/View/Layout/GeneratorPool.php

```diff
@@ -238,4 +247,19 @@ protected function visibilityConditionsExistsIn(array $data)
             array_key_exists('visibilityConditions', $data['attributes']) &&
             !empty($data['attributes']['visibilityConditions']);
     }
+
+    /**
+     * @param GeneratorInterface $generator
+     *
+     * @return bool
+     * @throws LocalizedException
+     */
+    private function validateGenerator($generator)
+    {
+        if (!($generator instanceof GeneratorInterface)) {
+            throw new LocalizedException(__('Generator classes must be instances of %1', GeneratorInterface::class));
```

Comment:
Do we really use `LocalizedException` in such cases? Please grep for some similar check, I believe it should be `InvalidArgumentException` and no need to translate.